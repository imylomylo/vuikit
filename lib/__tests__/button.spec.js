import { destroyVM, createVue, getSnapshot } from './_util'

describe('Button', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    const snapshot = getSnapshot(`
      <vk-button type="submit" color="primary" width="1-1" size="mini" :active="true" :disabled="true">
        Button
      </vk-button>
    `)
    expect(snapshot).toMatchSnapshot()
  })

  it('triggers event click when the button was clicked', () => {
    const vm = createVue(`<vk-button />`)
    const button = vm.$children[0]
    const cb = jest.fn()
    button.$on('click', cb)
    button.$el.click()
    expect(cb).toHaveBeenCalled()
    expect(cb.mock.calls[0].length).toBe(1)
  })
})
