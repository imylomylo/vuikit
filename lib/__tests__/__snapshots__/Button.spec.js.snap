exports[`Button renders correctly 1`] = `
"html
  body
    button.uk-button.uk-active.uk-button-default(type=\'submit\', disabled=\'disabled\', color=\'primary\', width=\'1-1\', size=\'mini\')
      | Button
"
`;
